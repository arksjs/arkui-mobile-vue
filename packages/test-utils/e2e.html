<script src="../../node_modules/vue/dist/vue.global.js"></script>
<script src="../../dist/index.js"></script>
<link rel="stylesheet" href="../../dist/index.css" />

<div id="app"></div>
