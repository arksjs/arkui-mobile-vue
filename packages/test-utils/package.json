{
  "name": "@arksjs/test-utils",
  "version": "0.0.1",
  "description": "arkui test-utils",
  "main": "dist/index.js",
  "private": true
}
