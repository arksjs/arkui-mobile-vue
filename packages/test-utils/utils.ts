export const timeout = (n: number) => new Promise(r => setTimeout(r, n))
