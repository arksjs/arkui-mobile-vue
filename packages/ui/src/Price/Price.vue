<template>
  <div class="fx-price">
    <slot></slot>
    <span class="fx-price_symbol" v-if="symbol">{{ symbol }}</span>
    <span class="fx-price_integer">{{ priceStr.split('.')[0] }}</span>
    <span class="fx-price_decimal" v-if="decimalDigits > 0"
      >.{{ priceStr.split('.')[1] }}</span
    >
  </div>
</template>

<script lang="ts">
import { defineComponent, computed } from 'vue'
import { isNumeric } from '../helpers/util'
import { getPrice } from './util'

export default defineComponent({
  name: 'fx-price',
  props: {
    // 金额
    price: {
      type: [Number, String],
      validator: isNumeric,
      default: 0,
      required: true
    },
    // 货币符号
    symbol: {
      type: String,
      default: null
    },
    // 是否千分位展示
    thousands: {
      type: Boolean,
      default: false
    },
    // 小数位数
    decimalDigits: {
      type: [Number, String],
      default: 2
    }
  },
  setup(props) {
    const priceStr = computed(() => getPrice(props))

    return {
      priceStr
    }
  }
})
</script>
