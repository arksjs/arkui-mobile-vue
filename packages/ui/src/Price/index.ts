import { withInstall } from '../helpers/with-install'
import Price from './Price.vue'

export { Price }
export const install = withInstall(Price)
export default Price
