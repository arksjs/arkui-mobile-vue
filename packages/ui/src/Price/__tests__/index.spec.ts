import { describe, test, expect } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import { Price } from '@/Price'

const price = 1111.11

describe('Price', () => {
  test.concurrent('snapshot', () => {
    const wrapper = shallowMount(Price, {
      props: {
        price
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('symbol="￥"', () => {
    const wrapper = shallowMount(Price, {
      props: {
        price,
        symbol: '￥'
      }
    })

    expect(wrapper.find('.ak-price_symbol').text()).toBe('￥')
  })

  test('thousands', () => {
    const wrapper = shallowMount(Price, {
      props: {
        price,
        thousands: true
      }
    })

    expect(wrapper.find('.ak-price_integer').text()).toBe('1,111')
  })

  test('decimalDigits=3', () => {
    const wrapper = shallowMount(Price, {
      props: {
        price,
        decimalDigits: 3
      }
    })

    expect(wrapper.find('.ak-price_decimal').text()).toBe('.110')
  })
})
