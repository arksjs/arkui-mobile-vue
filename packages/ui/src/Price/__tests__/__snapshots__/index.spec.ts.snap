// Vitest Snapshot v1

exports[`Price > snapshot 1`] = `
"<div class=\\"ak-price\\">
  <!--v-if--><span class=\\"ak-price_integer\\">1111</span><span class=\\"ak-price_decimal\\">.11</span>
</div>"
`;
